Compilation failure due to -Werror=format-overflow with g++ 8.1.0

Hey,

I was doing some spring cleaning in the applications directory that automatically compiles code and saw yours did not compile.

```
/home/vloup/applications/doom3/RBDOOM-3-BFG/doomclassic/doom/p_setup.cpp: In function 'void P_SetupLevel(int, int, int, skill_t)':
/home/vloup/applications/doom3/RBDOOM-3-BFG/doomclassic/doom/p_setup.cpp:661:22: error: '%i' directive writing between 2 and 10 bytes into a region of size 6 [-Werror=format-overflow=]
    sprintf (lumpname,"map%i", map);
                      ^~~~~~~
/home/vloup/applications/doom3/RBDOOM-3-BFG/doomclassic/doom/p_setup.cpp:661:22: note: directive argument in the range [10, 2147483647]
/home/vloup/applications/doom3/RBDOOM-3-BFG/doomclassic/doom/p_setup.cpp:661:12: note: 'sprintf' output between 6 and 14 bytes into a destination of size 9
    sprintf (lumpname,"map%i", map);
    ~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~
/home/vloup/applications/doom3/RBDOOM-3-BFG/doomclassic/doom/p_setup.cpp:659:22: error: '%i' directive writing between 1 and 11 bytes into a region of size 5 [-Werror=format-overflow=]
    sprintf (lumpname,"map0%i", map);
                      ^~~~~~~~
/home/vloup/applications/doom3/RBDOOM-3-BFG/doomclassic/doom/p_setup.cpp:659:22: note: directive argument in the range [-2147483648, 9]
/home/vloup/applications/doom3/RBDOOM-3-BFG/doomclassic/doom/p_setup.cpp:659:12: note: 'sprintf' output between 6 and 16 bytes into a destination of size 9
    sprintf (lumpname,"map0%i", map);
    ~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~
cc1plus: some warnings being treated as errors
make[2]: *** [CMakeFiles/RBDoom3BFG.dir/build.make:3898: CMakeFiles/RBDoom3BFG.dir/home/vloup/applications/doom3/RBDOOM-3-BFG/doomclassic/doom/p_setup.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:69: CMakeFiles/RBDoom3BFG.dir/all] Error 2
make: *** [Makefile:84: all] Error 2
```

Turns out that g++ is rather clever and infers the `map` value of this code over the full integer range, indeed bringing possible problems if `map` is negative or bigger than 5 digits in decimal representation:
```
                if (map<10)
                        sprintf (lumpname,"map0%i", map);
                else
                        sprintf (lumpname,"map%i", map);
```

Possible fixes:
1. Replace `map` with `map % 100` and force a 2 digit representation with `sprintf`'s format.
2. Actively check for `map` bounds (`map` >= 0, or convert to unsigned int and `map` < 100) and output an error otherwise.
3. Tweak the `CMakeLists.txt` file to exclude this error.
4. Increase `lumpname` array size so that no problems occur.

I let you pick what fits the best, that's why I didn't do any pull request.

Have a nice sunday.

Wanted to report that I'm running into this issue with gcc/g++ 8.2.0, output below.

[ 61%] Building CXX object CMakeFiles/RBDoom3BFG.dir/home/wls/git/RBDOOM-3-BFG/doomclassic/doom/p_saveg.cpp.o
[ 61%] Building CXX object CMakeFiles/RBDoom3BFG.dir/home/wls/git/RBDOOM-3-BFG/doomclassic/doom/p_setup.cpp.o
/home/wls/git/RBDOOM-3-BFG/doomclassic/doom/p_setup.cpp: In function ‘void P_SetupLevel(int, int, int, skill_t)’:
/home/wls/git/RBDOOM-3-BFG/doomclassic/doom/p_setup.cpp:661:22: error: ‘%i’ directive writing between 2 and 10 bytes into a region of size 6 [-Werror=format-overflow=]
    sprintf (lumpname,"map%i", map);
                      ^~~~~~~
/home/wls/git/RBDOOM-3-BFG/doomclassic/doom/p_setup.cpp:661:22: note: directive argument in the range [10, 2147483647]
/home/wls/git/RBDOOM-3-BFG/doomclassic/doom/p_setup.cpp:661:12: note: ‘sprintf’ output between 6 and 14 bytes into a destination of size 9
    sprintf (lumpname,"map%i", map);
    ~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~
/home/wls/git/RBDOOM-3-BFG/doomclassic/doom/p_setup.cpp:659:22: error: ‘%i’ directive writing between 1 and 11 bytes into a region of size 5 [-Werror=format-overflow=]
    sprintf (lumpname,"map0%i", map);
                      ^~~~~~~~
/home/wls/git/RBDOOM-3-BFG/doomclassic/doom/p_setup.cpp:659:22: note: directive argument in the range [-2147483648, 9]
/home/wls/git/RBDOOM-3-BFG/doomclassic/doom/p_setup.cpp:659:12: note: ‘sprintf’ output between 6 and 16 bytes into a destination of size 9
    sprintf (lumpname,"map0%i", map);
    ~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~
cc1plus: some warnings being treated as errors
make[2]: *** [CMakeFiles/RBDoom3BFG.dir/build.make:3898: CMakeFiles/RBDoom3BFG.dir/home/wls/git/RBDOOM-3-BFG/doomclassic/doom/p_setup.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:74: CMakeFiles/RBDoom3BFG.dir/all] Error 2
make: *** [Makefile:84: all] Error 2

--- a/doomclassic/doom/p_setup.cpp
+++ b/doomclassic/doom/p_setup.cpp
@@ -656,9 +656,9 @@ P_SetupLevel
 	if ( ::g->gamemode == commercial)
 	{
 		if (map<10)
-			sprintf (lumpname,"map0%i", map);
+			sprintf (lumpname, "map0%i", map % 100);
 		else
-			sprintf (lumpname,"map%i", map);
+			sprintf (lumpname, "map%i", map % 100);
 	}
 	else
 	{

--- a/neo/renderer/RenderBackend.cpp
+++ b/neo/renderer/RenderBackend.cpp
@@ -435,7 +435,7 @@ void RB_SetVertexColorParms( stageVertexColor_t svc )
 RB_GetShaderTextureMatrix
 ======================
 */
-static void RB_GetShaderTextureMatrix( const float* shaderRegisters, const textureStage_t* texture, float matrix[16] )
+void RB_GetShaderTextureMatrix( const float* shaderRegisters, const textureStage_t* texture, float matrix[16] )
 {
 	matrix[0 * 4 + 0] = shaderRegisters[ texture->matrix[0][0] ];
 	matrix[1 * 4 + 0] = shaderRegisters[ texture->matrix[0][1] ];
@@ -474,7 +474,7 @@ static void RB_GetShaderTextureMatrix( const float* shaderRegisters, const textu
 RB_LoadShaderTextureMatrix
 ======================
 */
-static void RB_LoadShaderTextureMatrix( const float* shaderRegisters, const textureStage_t* texture )
+void RB_LoadShaderTextureMatrix( const float* shaderRegisters, const textureStage_t* texture )
 {
 	float texS[4] = { 1.0f, 0.0f, 0.0f, 0.0f };
 	float texT[4] = { 0.0f, 1.0f, 0.0f, 0.0f };
@@ -509,7 +509,7 @@ static void RB_LoadShaderTextureMatrix( const float* shaderRegisters, const text
 RB_BakeTextureMatrixIntoTexgen
 =====================
 */
-static void RB_BakeTextureMatrixIntoTexgen( idPlane lightProject[3], const float* textureMatrix )
+void RB_BakeTextureMatrixIntoTexgen( idPlane lightProject[3], const float* textureMatrix )
 {
 	float genMatrix[16];
 	float final[16];

--- a/neo/sys/sdl/sdl_events.cpp
+++ b/neo/sys/sdl/sdl_events.cpp
@@ -40,7 +40,7 @@ If you have questions concerning this license or the applicable additional terms
 
 #include <SDL.h>
 
-#include "renderer/tr_local.h"
+#include "renderer/RenderCommon.h"
 #include "sdl_local.h"
 #include "../posix/posix_public.h"
 

--- a/neo/sys/sdl/sdl_glimp.cpp
+++ b/neo/sys/sdl/sdl_glimp.cpp
@@ -40,7 +40,7 @@ If you have questions concerning this license or the applicable additional terms
 
 #include <SDL.h>
 
-#include "renderer/tr_local.h"
+#include "renderer/RenderCommon.h"
 #include "sdl_local.h"
 
 idCVar in_nograb( "in_nograb", "0", CVAR_SYSTEM | CVAR_NOCHEAT, "prevents input grabbing" );

